service/worker/deployment/fx.go

```diff
@@ -0,0 +1,125 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2024 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package deployment
+
+import (
+	"fmt"
+	"log"
+
+	enumspb "go.temporal.io/api/enums/v1"
+	"go.temporal.io/api/workflowservice/v1"
+	sdkworker "go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+	"go.temporal.io/server/api/matchingservice/v1"
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/metrics"
+	"go.temporal.io/server/common/namespace"
+	"go.temporal.io/server/common/sdk"
+	"go.temporal.io/server/common/searchattribute"
+	workercommon "go.temporal.io/server/service/worker/common"
+	"go.uber.org/fx"
+)
+
+const (
+	WorkflowType      = "temporal-sys-deployment-workflow"
+	NamespaceDivision = "TemporalDeployment"
+)
+
+var (
+	VisibilityBaseListQuery = fmt.Sprintf(
+		"%s = '%s' AND %s = '%s' AND %s = '%s'",
+		searchattribute.WorkflowType,
+		WorkflowType,
+		searchattribute.TemporalNamespaceDivision,
+		NamespaceDivision,
+		searchattribute.ExecutionStatus,
+		enumspb.WORKFLOW_EXECUTION_STATUS_RUNNING.String(),
+	)
+)
+
+type (
+	workerComponent struct {
+		activityDeps             activityDeps
+		enabledForNs             dynamicconfig.BoolPropertyFnWithNamespaceFilter
+		globalNSStartWorkflowRPS dynamicconfig.TypedSubscribableWithNamespaceFilter[float64]
+		maxBlobSize              dynamicconfig.IntPropertyFnWithNamespaceFilter
```

Comment:
I don't think you need this :)

Reply:
I think we do need `activityDeps` as well as `enabledForNS` though. From the top of my head, the versioning worker component would have activity dependencies since we would most certainly require the matching client for making RPC's. `enabledForNS` would serve as a flag for toggling this on/off. 

`maxBlobSize` is something I don't see we would need so I will remove that. Unless I have missed the point, I would like to know why we would not need `activityDeps` and  `enabledForNS` for the worker component.

Reply:
My comment was only for the line it said it was for

Reply:
For some reason, my github shows you were reviewing multiple lines (line 64 - 68 in this "outdated" snippet). Hence, I thought you suggested to remove `activityDeps` & `enabledForNs`. If you intended to only remove the other two, I have done so!